innative/atomic_instruction_details.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c)2020 Black Sphere Studios
+// For conditions of distribution and use, see copyright notice in innative.h
+
+#pragma once
+
+namespace innative::atomic_details {
+  constexpr auto OP_START      = OP_i32_atomic_load;
+  constexpr auto OP_END        = OP_atomic_end;
+  constexpr auto OP_GROUP_SIZE = 7;
+  enum class OpGroup
+  {
+    Load,
+    Store,
+    Add,
+    Sub,
+    And,
+    Or,
+    Xor,
+    Xchg,
+    CmpXchg,
+
+    INVALID_CAT,
+  };
+
+  constexpr OpGroup GetOpGroup(uint8_t op) { return OpGroup((op - OP_START) / OP_GROUP_SIZE); }
+  constexpr int GetOpType(uint8_t op) { return (op - OP_START) % OP_GROUP_SIZE; }
+  constexpr uint8_t IsI64(int opType) { return uint8_t(0x01'01'01'00'00'01'00 >> (opType * 8)); }
```

Comment:
Can a comment be left here explaining what the heck this is doing, and what the byte separations in the constant are for?